doc: suit: Push scenario and DFU caches documentation (#17519)
nrfconnect/sdk-nrf: samples/suit/smp_transfer/README.rst

If you want to make modifications to how the DFU is executed in this sample, you can do so by editing the manifest templates, or generating your own custom manifests.
See the :ref:`ug_nrf54h20_suit_customize_dfu` user guide for instructions and examples.

.. _nrf54h_suit_sample_extflash:

External flash support
======================

> You can build the application with external flash support by running:

Reviewer: The code is missing.

Author: Should be fixed now

.. code-block:: console

> From the sample directory.

Reviewer: Should line 120 be combined with 115, seems bit incomplete?

Author: Yes, fixed this

>    west build -p -b nrf54h20dk/nrf54h20/cpuapp -- -DFILE_SUFFIX="extflash"

Reviewer 2: I would not remove "pristine" build option, this helps avoid build error and it is used in "Building And Running" section here, pristine should not be used when building new project version to speed up building, but for first build example command with pristine is better. That is also for line 128 here.

With this configuration, the sample is configured to use UART as the transport and the external flash is enabled.

> To see which Kconfig options are needed to achieve that, see the ``sample.suit.smp_transfer.cache_push.extflash`` configuration in :file:`samples/suit/sample.yaml`.

Reviewer: ```suggestion
To see which Kconfig options are needed to achieve that, see the ``sample.suit.smp_transfer.cache_push.extflash`` configuration in the :file:`samples/suit/sample.yaml` file.
```

To enable both the external flash and the BLE transport, use the following command:

.. code-block:: console

   west build ./ -b nrf54h20dk/nrf54h20/cpuapp -T  sample.suit.smp_transfer.cache_push.extflash.bt

.. note::

>    This way of building the application with external flash will enable the "push" scenario for updating from external flash.

Reviewer: Could it be rephrased like this?
"Building the application with external flash will enable the push scenario for updating from external flash."

Author: Not really, it is about this specific way of building. You can also build in a way which will enable the fetch scenario.
Shortened this to:
"This way of building the application will enable the "push" scenario for updating from external flash"
hopefully it makes it a bit clearer

>    This way of building the application will enable the "push" scenario for updating from external flash.

Reviewer: ```suggestion
   This way of building the application will enable the push scenario for updating from external flash.
```

   It will also extract the image to a DFU cache partition file.

>    For more information, see :ref:`How to push SUIT payloads to multiple partitions <ug_nrf54h20_suit_push>`

Reviewer: ```suggestion
   For more information, see :ref:`How to push SUIT payloads to multiple partitions <ug_nrf54h20_suit_push>`.
```

               197.40 KiB / 244.57 KiB [==============================================================================================================================>------------------------------]  80.71% 20.51 KiB/s 00m02s
               241.16 KiB / 244.57 KiB [=================================================================================================================================================================>--]  98.60% 20.74 KiB/s
               Done

>       #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash using:

Reviewer 3: ```suggestion
      #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash using the following command:
```

Author: Fixed

            .. code-block:: console

               mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image upload -n 2 build/DFU/root.suit

>             Note the ``-n 2`` parameter - this means uploading to DFU cache partition 1 (image 0 is the envelope, image 1 is the cache partition 0).

Reviewer 3: ```suggestion
            .. note:: 
               the ``-n 2`` parameter uploads to DFU cache partition 1 (where image 0 is the envelope and image 1 is the cache partition 0).
```

Author: Ok, fixed

>       #. Start the installaltion of the new firmware using:

Reviewer 3: ```suggestion
      #. Start the installaltion of the new firmware as follows:
```

Author: Fixed

>       #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash using:
> 
>             .. code-block:: console
> 
>                mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image upload -n 2 build/DFU/root.suit
> 
>             Note the ``-n 2`` parameter - this means uploading to DFU cache partition 1 (image 0 is the envelope, image 1 is the cache partition 0).
> 
>       #. Start the installaltion of the new firmware using:
> 
>             .. code-block:: console
> 
>                mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image confirm

Reviewer: ```suggestion
      #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash using:

         .. code-block:: console

            mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image upload -n 2 build/DFU/root.suit

         The ``-n 2`` parameter refers to the uploading to DFU cache partition 1 (image 0 is the envelope, image 1 is the cache partition 0).

      #. Start the installaltion of the new firmware using:

         .. code-block:: console

            mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image confirm
```

Reviewer 3: ```suggestion
      #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash as follows:

            .. code-block:: console

               mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image upload -n 2 build/DFU/root.suit

            Note the ``-n 2`` parameter - this means uploading to DFU cache partition 1 (image 0 is the envelope, image 1 is the cache partition 0).

      #. Start the installaltion of the new firmware as follows:

            .. code-block:: console

               mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image confirm
```

               197.40 KiB / 244.57 KiB [==============================================================================================================================>------------------------------]  80.71% 20.51 KiB/s 00m02s
               241.16 KiB / 244.57 KiB [=================================================================================================================================================================>--]  98.60% 20.74 KiB/s
               Done
      #. If you have built the application with :ref:`external flash support <nrf54h_suit_sample_extflash>`, upload the cache partition to the external flash using the following command:

         .. code-block:: console

            mcumgr --conntype serial --connstring "dev=/dev/ttyACM0,baud=115200" image upload -n 2 build/DFU/dfu_cache_partition_1.bin

            .. note::
               the ``-n 2`` parameter uploads to DFU cache partition 1 (where image 0 is the envelope and image 1 is the cache partition 0).

>       #. Start the installaltion of the new firmware as follows:

Reviewer 4: *installation
